doc/source/geotransformations.md

```diff
@@ -0,0 +1,204 @@
+rast_reproj = rast.reproject(
+    res=(rast.res[0] * 2, rast.res[1] / 2),
+    bounds={"left": rast.bounds.left, "bottom": rast.bounds.bottom, 
+            "right": rast.bounds.left + 10000, "top": rast.bounds.bottom + 10000},
+    resampling="cubic")
+```
+
+```{code-cell} ipython3
+:tags: [hide-input]
+:mystnb:
+:  code_prompt_show: "Show the code for plotting the figure"
+:  code_prompt_hide: "Hide the code for plotting the figure"
+
+f, ax = plt.subplots(1, 2)
+ax[0].set_title("Before reprojection")
+rast.plot(ax=ax[0])
+vect.plot(rast, ax=ax[0], ec="k", fc="none")
+ax[1].set_title("After reprojection")
+rast_reproj.plot(ax=ax[1])
+vect_reproj.plot(ax=ax[1], ec="k", fc="none")
+_ = ax[1].set_yticklabels([])
+```
+
+```{note}
+In GeoUtils, `"bilinear"` is the default resampling method. A simple {class}`str` matching the naming of a {class}`rasterio.enums.Resampling` method can be
+passed.
+
+Resampling methods are listed in **[the dedicated section of Rasterio's API](https://rasterio.readthedocs.io/en/latest/api/rasterio.enums.html#rasterio.enums.Resampling)**.
+```
+
+We can also simply pass another raster as reference to reproject to match the same CRS, and re-grid to the same bounds 
+and resolution:
+
+```{code-cell} ipython3
+# Reproject vector to CRS of raster by simply passing the raster
+rast_reproj2 = rast.reproject(rast2)
+```
+
+As shown above, if the rasters have different {ref}`Pixel interpretation<pixel-interpretation>`, GeoUtils 
+raises a warning to ensure this is intended. 
+This warning can be turned off at the package-level using [GeoUtils' global configuration]().
+
+```{code-cell} ipython3
+:tags: [hide-input]
+:mystnb:
+:  code_prompt_show: "Show the code for plotting the figure"
+:  code_prompt_hide: "Hide the code for plotting the figure"
+
+f, ax = plt.subplots(1, 3)
+ax[0].set_title("Raster 1")
+rast.plot(ax=ax[0])
+vect.plot(rast, ax=ax[0], ec="k", fc="none")
+ax[1].set_title("Raster 2")
+rast2.plot(ax=ax[1], cmap="Reds",)
+vect.plot(rast, ax=ax[1], ec="k", fc="none")
+ax[2].set_title("Match-ref\nreprojection")
+rast_reproj2.plot(ax=ax[2])
+vect_reproj.plot(ax=ax[2], ec="k", fc="none")
+_ = ax[1].set_yticklabels([])
+_ = ax[2].set_yticklabels([])
+```
+
+## Crop or pad
+
+{func}`geoutils.Raster.crop` or {func}`geoutils.Vector.crop`.
+
+Cropping **modifies the spatial bounds of the geospatial data in a rectangular extent**, by removing or adding data 
+(in which case it corresponds to padding) without resampling. For this, we use {func}`~geoutils.Raster.crop`.
+
+For rasters, cropping removes or adds pixels to the sides of the raster grid.
+
+For vectors, cropping removes some geometry features around the bounds, with three options possible:
+1. Removing all features **not intersecting** the cropping geometry,
+2. Removing all features **not contained** in the cropping geometry,
+3. Making all features **exactly clipped** to the cropping geometry (modifies the geometry data).
```

Comment:
This made me think, we should have the same functionality for Rasters. Added an issue: https://github.com/GlacioHack/geoutils/issues/583

Reply:
Good point!